### DISCUSSION TITLE
Github Package snapshot build number not updating

### DISCUSSION THREAD

**Discussion by Derongan** [community/community#24658] [Apps, API and Webhooks]
<p>When I publish new versions of a SNAPSHOT build to github packages, the build number is not updated.</p>
<pre><code>&lt;metadata modelVersion=""&gt;
  &lt;groupId&gt;com.derongan.minecraft&lt;/groupId&gt;
  &lt;artifactId&gt;guiy&lt;/artifactId&gt;
  &lt;version&gt;0.1.0-SNAPSHOT&lt;/version&gt;
  &lt;versioning&gt;
    &lt;snapshot&gt;
    &lt;timestamp&gt;20200305.031042&lt;/timestamp&gt;
    &lt;buildNumber&gt;1&lt;/buildNumber&gt;
  &lt;/snapshot&gt;
  &lt;snapshotVersion&gt;
    &lt;extension&gt;jar&lt;/extension&gt;
    &lt;value&gt;0.1.0-20200305.034406-2&lt;/value&gt;
    &lt;updated&gt;20200305034436&lt;/updated&gt;
  &lt;/snapshotVersion&gt;
  &lt;snapshotVersion&gt;
    &lt;extension&gt;jar&lt;/extension&gt;
    &lt;value&gt;0.1.0-20200305.190209-2&lt;/value&gt;
    &lt;updated&gt;20200305190240&lt;/updated&gt;
  &lt;/snapshotVersion&gt;
</code></pre>
<p>I think this is causing gradle to always download build-number 1, and any later snapshots are ignored.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/derongan">@derongan</a>, as far as I know we don’t support mutating packages in terms of publishing a new SNAPSHOT to override an existing SNAPSHOT.</p>

**Comment by djweber** [comment]
<p><a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>  Just to clarify, does this mean that we cannot upload multiple builds for the same snapshot (e.g. 1.0.0-SNAPSHOT) and have GitHub serve the latest artifact? Encountering a similar issue at the moment where only the first build is being pulled down from the repository (even after deleting my local cache). </p>

**Comment by jcansdale** [comment, marked as answer]
<p>We do have partial support for SNAPSHOT versions. You can push snapshot packages that contain plain .jar and .pom files. There is however a bug that stops it working with -javadoc and -source artifacts. This is somethign to be aware of because it trips up lots of people! Hopefully this will be fixed soon.</p>

**Comment by sbabych** [comment]
<p><a class="mention" href="/jcansdale">@jcansdale</a> </p>
<p>Hi, we publish snapshot packages with Gradle, it’s a plain .pom and .jar files. But unfortunately when we try to use snapshot package only first uploaded jar is downloaded. </p>
<p>Also did try to publish packages with new Gradle module metadata files - and it’s the same, only first uploaded jar is downloaded. Do you know when snapshots will work in GitHub Packages ? thank you </p>

**Comment by sbabych** [comment]
<p><a class="mention" href="/derongan">@derongan</a> did you resolve you problem ? thank you </p>

**Comment by jcansdale** [comment]
<p>Could you open a ticket here with a link to your repository any logs for publishing/consuming your package?</p>
<p><a href="https://support.github.com/contact" rel="nofollow noopener">https://support.github.com/contact</a></p>
<p>Thanks,</p>
<p>Jamie.</p>

**Comment by nick0602** [comment]
<p>Currently experiencing the same issue on a Gradle project, for some reason when multiple snapshots of the same version are uploaded (e.g. 1.0.0-SNAPSHOT), even when the cache for changing modules is disabled - explicitly set it to 0 seconds - GH Packages always returns the first artifact.</p>
<p>I thought it was a Gradle misconfiguration I made at first, but I can see in the logs that the resolved pom file is the one belonging to the first upload and any other dependency pulled from other Maven repos (not GH packages ones) with the same cache policy is always the latest.</p>
<p>Any chance this can get fixed? Thanks!</p>

**Comment by nick0602** [comment]
<p>I opened a support ticket and got a confirmation today that the GH team is currently looking into the issue of the SNAPSHOT always returning the first version. I’ll post here again when I get updates.</p>

**Comment by jcansdale** [comment]
<p>Thanks for opening the ticket. A new version was deployed earlier today. This issue should now be fixed!</p>

**Comment by nick0602** [comment]
<p>Works great now, thanks again!</p>

**Comment by nickmelis** [comment]
<p>Having the same problem here. Trying to publish a 0.0.1-SNAPSHOT artifact with Gradle. No error on the publishing side, but there’s no trace of the most recent artifact on GitHub. How do I solve this please?</p>

**Comment by akash-findeed** [comment]
<p>Have the same problem here. Is this getting solved any time soon?</p>

**Comment by saiintbrisson** [comment]
<p>Still having the same issue, really frustrating as snapshots are really important for our development environment. I hope we hear back from the staff soon, switching back to our old repo in the meanwhile.</p>

**Comment by nickmelis** [comment]
<p>Yes, we had to move to Artifactory for that reason, which is a shame.</p>

**Comment by jordansilva** [comment]
<p>Same here. First I thought it was a mistake from my side, but after a couple of hours debugging it I can confirm the SNAPSHOT version is uploaded to Github Packages successfully but I can’t find it there or fetch the latest version.<br>
I’m also using Gradle.</p>

**Comment by reinhardbluelab** [comment]
<p>I managed to get it to work. A Maven project is publishing the package and I can see the new SNAPSHOT version on the GitHub package page under the <em>Assets</em> listing. The timestamps on the GitHub repo do not seem to change, which is a bit weird.</p>
<p>A Gradle project is importing the SNAPSHOT package, but do ensure that Gradle is not caching changing dependencies. Otherwise it might not download the latest version (even if you build with <code>--refresh-dependencies</code>). Edit <em>gradle.build</em> to include this:</p>
<p><code>configurations.all { resolutionStrategy.cacheChangingModulesFor 0, 'seconds' }</code></p>
<p>See <a href="https://stackoverflow.com/a/14804849/475511" rel="noopener nofollow ugc">https://stackoverflow.com/a/14804849/475511</a>.</p>
